cinema/tests/test_movie_api.py

```diff
@@ -157,3 +158,195 @@ def test_image_url_is_shown_on_movie_session_detail(self):
+        serializer_with_genre_1 = MovieListSerializer(movie_with_genre_1)
+        serializer_with_genre_2 = MovieListSerializer(movie_with_genre_2)
+
+        self.assertNotIn(serializer_without_genres.data, result.data)
+        self.assertIn(serializer_with_genre_1.data, result.data)
+        self.assertIn(serializer_with_genre_2.data, result.data)
+
+    def test_filter_movies_by_title(self):
+        movie_1 = sample_movie(title="Interstellar")
+        movie_2 = sample_movie(title="Sherlock Holmes")
+
+        result = self.client.get(
+            MOVIE_URL, {"title": "Interstellar"}
+        )
+
+        serializer_movie_1 = MovieListSerializer(movie_1)
+        serializer_movie_2 = MovieListSerializer(movie_2)
+
+        self.assertIn(serializer_movie_1.data, result.data)
+        self.assertNotIn(serializer_movie_2.data, result.data)
+
+    def test_retrieve_movie_detail(self):
+        movie = sample_movie()
+
+        movie.actors.add(sample_actor())
+        movie.genres.add(sample_genre())
+
+        url = detail_url(movie.id)
+
+        result = self.client.get(url)
+
+        serializer = MovieDetailSerializer(movie)
+
+        self.assertEqual(result.status_code, status.HTTP_200_OK)
+        self.assertEqual(result.data, serializer.data)
+
+    def test_create_movie_forbidden(self):
+        payload = {
+            "title": "Interstellar",
+            "description": "si-fi",
+            "duration": 150,
+        }
+
+        result = self.client.post(MOVIE_URL, payload)
+
+        self.assertEqual(result.status_code, status.HTTP_403_FORBIDDEN)
+
+
+class AdminMovieTests(TestCase):
+
+    def setUp(self):
+        self.client = APIClient()
+        self.user = get_user_model().objects.create_user(
+            email="superadmin@test.test",
+            password="super_password54321",
+            is_staff=True
+        )
+        self.client.force_authenticate(self.user)
+
+    def test_create_movie(self):
+        genre_1 = sample_genre()
+        actor_1 = sample_actor()
+        payload = {
+            "title": "Shutter Island",
+            "description": "Very interesting movie",
+            "duration": 80,
+            "genres": [genre_1.id,],
+            "actors": [actor_1.id,],
+        }
+
+        result = self.client.post(MOVIE_URL, payload, format="json")
+
+        self.assertEqual(result.status_code, status.HTTP_201_CREATED)
+
+        movie = Movie.objects.get(id=result.data["id"])
+
+
+
+        self.assertEqual(movie.title, payload["title"])
+        self.assertEqual(movie.description, payload["description"])
+        self.assertEqual(movie.duration, payload["duration"])
+
+
+    def test_create_movie_with_actor_and_genre(self):
+        actor = sample_actor()
+        genre = sample_genre()
+
+        payload = {
+            "title": "Shutter Island",
+            "description": "Very interesting movie",
+            "duration": 80,
+            "actors": actor.id,
+            "genres": genre.id
```

Comment:
In the `test_create_movie_with_actor_and_genre` method, the `actors` and `genres` fields in the payload should be lists (e.g., `"actors": [actor.id]` and `"genres": [genre.id]`) to match the expected input format for these fields.